fix for crop support in cellfinder

parikshit14 opened the pull request:
Before submitting a pull request (PR), please read the [contributing guide](https://github.com/brainglobe/.github/blob/main/CONTRIBUTING.md).

Please fill out as much of this template as you can, but if you have any problems or questions, just leave a comment and we will help out :)

## Description
Workflow test failing in cellfinder after adding crop support for the given cell width and height.
Particularly: `tests/brainmapper/test_integration/test_detection.py`

**What is this PR**

- [x] Bug fix
- [ ] Addition of a new feature
- [ ] Other

**Why is this PR needed?**

**What does this PR do?**

## References
Issue: https://github.com/brainglobe/cellfinder/issues/468 
PR: https://github.com/brainglobe/cellfinder/pull/494


Please reference any existing issues/PRs that relate to this PR.

## How has this PR been tested?
Yes, tested locally on my version of cellfinder (with changes of crop support).

Please explain how any new code has been tested, and how you have ensured that no existing functionality has changed.

## Is this a breaking change?
Will break, but since the changes will eventually be merged in cellfinder, it will be ok. Currently in a deadlock situation, i.e. both places' tests will fail untill one gets merged

If this PR breaks any existing functionality, please explain how and why.

## Does this PR require an update to the documentation?

If any features have changed, or have been added. Please explain how the documentation has been updated (and link to the associated PR). See [here](https://brainglobe.info/community/developers/index.html#to-improve-the-documentation) for details.

## Checklist:

- [x] The code has been tested locally
- [x] Tests have been added to cover all new functionality (unit & integration)
- [x] The documentation has been updated to reflect any changes
- [x] The code has been formatted with [pre-commit](https://pre-commit.com/)

adamltyson commented:
Hi @parikshit14, is there no way to fix this on the cellfinder side, i.e. make it backwards compatible? I'd rather not add more CLI args to brainmapper if we can help it.

parikshit14 commented:
I dont think it is possible @adamltyson , since https://github.com/brainglobe/brainglobe-workflows/blob/0dac3faa35e9794535af4514aecb93594f44fd9a/tests/brainmapper/test_integration/test_detection.py#L32 defines args.

adamltyson commented:
But in the API, the new arguments can be optional though?

parikshit14 commented:
Ya, you got a point. Let me check quickly if the order of declaring these parameters can help, maybe.

parikshit14 commented:
Updated the parameters sequence to pass the workflow modification. The sequence indeed helped. 
Tested locally and works well. Let me know, I'll close this PR then.

alessandrofelder reviewed (commented):
Please make sure the tests pass in the CI system as well @parikshit14, then someone can review

parikshit14 commented:
Hi @alessandrofelder , I have mentioned the reasons for failing.

> Issue: [brainglobe/cellfinder#468](https://github.com/brainglobe/cellfinder/issues/468)
> PR: [brainglobe/cellfinder#494](https://github.com/brainglobe/cellfinder/pull/494)

Please refer to these. I think @adamltyson was looking at this and suggested making changes in the referenced PR  instead of adding the crop parameters in the workflow. 

The reason I added parameters was that there was a test from this repository failing in my updated code in CellFinder. Later, I changed the positions of these parameters and the tests were passing, awaiting for approval in cellfinder.

alessandrofelder commented:
Ah, cool, thanks for clarifying!

adamltyson commented:
Hi @parikshit14, now the tests in https://github.com/brainglobe/cellfinder/pull/494 are passing, can this PR now be closed?

adamltyson commented:
I'm going to close this as I think it's no longer needed. Feel free to reopen if necessary.
